packages/e2e-tests/config/setup-test-framework.js

```diff
@@ -187,28 +187,33 @@ async function simulateAdverseConditions() {
 		return;
 	}
 
-	const client = await page.target().createCDPSession();
+	if ( OFFLINE ) {
+		await page.setOfflineMode( true );
+		await page.emulateNetworkConditions( {
+			// Disable download/upload.
+			download: -1,
+			upload: -1,
+			// Disable latency.
+			latency: 0,
+		} );
+	}
 
-	if ( SLOW_NETWORK || OFFLINE ) {
+	if ( SLOW_NETWORK ) {
 		// See: https://chromedevtools.github.io/devtools-protocol/tot/Network#method-emulateNetworkConditions
 		// The values below simulate fast 3G conditions as per https://github.com/ChromeDevTools/devtools-frontend/blob/80c102878fd97a7a696572054007d40560dcdd21/front_end/sdk/NetworkManager.js#L252-L274
-		await client.send( 'Network.emulateNetworkConditions', {
-			// Network connectivity is absent
-			offline: Boolean( OFFLINE || false ),
+		await page.emulateNetworkConditions( {
 			// Download speed (bytes/s)
-			downloadThroughput: ( ( 1.6 * 1024 * 1024 ) / 8 ) * 0.9,
+			download: ( ( 1.6 * 1024 * 1024 ) / 8 ) * 0.9,
 			// Upload speed (bytes/s)
-			uploadThroughput: ( ( 750 * 1024 ) / 8 ) * 0.9,
+			upload: ( ( 750 * 1024 ) / 8 ) * 0.9,
 			// Latency (ms)
 			latency: 150 * 3.75,
 		} );
 	}
```

Comment:
I think we can just use the predefined conditions here?

```js
await page.emulateNetworkConditions( puppeteer.networkConditions['Fast 3G'] );
```

Then, we can remove the comments above too.

Reply:
This might seem silly, but the puppeteer object can't be imported.

Neither of these seem to work:
```
import puppeteer from 'puppeteer';
```
```
import puppeteer from 'puppeteer-core';
```

Not sure if you have a quick fix, but I don't want to mess around too much with dependencies, much more important stuff to do, so I think let's just ship this as is for now. Or feel free to push a commit if you can solve it.

Reply:
Hmm, seems like a npm/eslint issue. Peer dependency should be installed by default. Upgrading to npm 7 might fix that. But sure, this is good enough 👍.